static/app/views/performance/newTraceDetails/index.tsx

```diff
@@ -64,7 +65,18 @@ export function TraceView() {
   const params = useParams<{traceSlug?: string}>();
   const organization = useOrganization();
 
-  const traceSlug = useMemo(() => params.traceSlug?.trim() ?? '', [params.traceSlug]);
+  const traceSlug = useMemo(() => {
+    const slug = params.traceSlug?.trim() ?? '';
+    // null and undefined are not valid trace slugs, but they can be passed
+    // in the URL and need to check for their string coerced values.
+    if (!slug || slug === 'null' || slug === 'undefined') {
+      Sentry.withScope(scope => {
+        scope.setFingerprint(['trace-null-slug']);
+        Sentry.captureMessage(`Trace slug is empty, got ${slug}`);
```

Comment:
Is setting the slug in the message useful if it's always one of empty string, null or undefined?

Reply:
yeah, let me remove it, we should have enough info